### PR TITLE
handle error safely in healthcheck

### PR DESCRIPTION
**Pull request by deitch**
Due to how the healthcheck logic worked, it tried to do `rest.Body.Close()` even if `resp.Body` was `nil`. This ensures we only close if `err == nil && `resp.Body != nil`.

Thanks to @displague for figuring out the logic problem.

### CODE REVIEW

File: metal/eip_controlplane_reconciliation.go [hunk 1]
```diff
@@ -106,9 +106,13 @@ func (m *controlPlaneEndpointManager) reconcileNodes(ctx context.Context, nodes
 		return err
 	}
 	resp, err := m.httpClient.Do(req)
+	// if there was no error, ensure we close
+	if err == nil && resp.Body != nil {
```
Comment (displague) [inline metal/eip_controlplane_reconciliation.go:110]: sounds like the `resp.Body` check may not be needed here:
https://stackoverflow.com/a/33238755/156674
https://golang.org/pkg/net/http/#Client.Do
> If the returned error is nil, the Response will contain a non-nil Body which the user is expected to close.

File: metal/eip_controlplane_reconciliation.go [hunk 2]
```diff
@@ -106,9 +106,13 @@ func (m *controlPlaneEndpointManager) reconcileNodes(ctx context.Context, nodes
...
+		defer resp.Body.Close()
+	}
 	if err != nil || resp.StatusCode != http.StatusOK {
 		if err != nil {
-			klog.Errorf("http client error during healthcheck. err \"%s\"", err)
+			klog.Errorf("http client error during healthcheck, will try to reassign to a healthy node. err \"%s\"", err)
```
Comment (displague) [inline metal/eip_controlplane_reconciliation.go:115]: `\"%s\"` could be `%q`

**Review by displague — APPROVED** [review]
the logging and defer changes make sense (I didn't look outside the box of this PR).

File: metal/eip_controlplane_reconciliation.go (hunk 1 above)
Comment (deitch) [reply metal/eip_controlplane_reconciliation.go:110]: You are right, but it doesn't hurt.

**Pull request merged**